### PR TITLE
Vimc 3200 - FOURTH - touchstone_country.csv support

### PR DESCRIPTION
**Pull request by weshinsley**
The next round of bliss, which adds touchstone_country support...

A few refactors in this: where possible, I've moved the code for checking whether touchstones are in-prep (ie, whether we can fiddle with them) into touchstone:load_touchstone, at which point the db transcation will include any new touchstones added. There were also a few stray mashes still being done which didn't call the mash function, and utils:assign_serial_ids I found I had rewritten mostly a couple of times.

### CODE REVIEW

File: DESCRIPTION [hunk 1]
```diff
@@ -1,12 +1,12 @@
 Package: stoner
 Title: Support for building VIMC Montagu Touchstones, using dettl
-Version: 0.0.3
+Version: 0.0.4
 Authors@R:
-    c(person("Wes", "Hinsley",role = c("aut", "cre", "dnc", "itr"),
+    c(person("Wes", "Hinsley",role = c("aut", "cre", "dnc", "itr", "sng"),
```
Comment (richfitz) [inline DESCRIPTION:5]: :man_singer:

File: R/touchstone_countries.R [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    return(list())
+  }
+
+  diseases <- lapply(csv$diseases, split_semi)
+  countries <- lapply(csv$countries, split_semi)
+
+  # Build a table that multiplies out countries and diseases...
+  # for each touchstone.
+
+  expand_countries <- unlist(lapply(1:nrow(csv), function(x) {
```
Comment (richfitz) [inline R/touchstone_countries.R]: small thing, but can we avoid `1:nrow(csv)` here please?

File: tests/testthat/examples/tc/add_t_app_zero/meta/db_disease.csv [hunk 3]
```diff
@@ -0,0 +1,2 @@
+id,name
+flu,Elf Flu
```
Comment (richfitz) [inline tests/testthat/examples/tc/add_t_app_zero/meta/db_disease.csv:2]: getting the AoC spirit on early I see

**Review by richfitz — CHANGES_REQUESTED** [review]
one tiny thing then let's get this moving again...

File: tests/testthat/examples/tc/add_t_app_zero/meta/db_disease.csv (hunk 3 above)
Comment (weshinsley) [reply tests/testthat/examples/tc/add_t_app_zero/meta/db_disease.csv:2]: :santa:

File: R/touchstone_countries.R (hunk 2 above)
Comment (weshinsley) [reply R/touchstone_countries.R]: Done - good spot. (Would have failed on country CSV with a header but no rows)

**Review by richfitz — APPROVED** [review]

**Pull request merged**